src/client/fuse_ll.cc

```diff
@@ -163,6 +167,37 @@ class CephFuse::Handle {
   struct fuse_args args;
 };
 
+static bool already_mounted(const char *path)
+{
+  FILE *mtab = NULL;
+  struct mntent *mnt;
+  bool retval = false;
+
+  // get mount information from /etc/mtab
+  mtab = setmntent(_PATH_MOUNTED, "r");
+  if (mtab == NULL) {
+    int err = errno;
+    derr << "fuse_ll: already_mounted: cannot open " << _PATH_MOUNTED
+      << ". Failed with error " << cpp_strerror(err) << dendl;
+    return retval;
+  }
+
+  // check whether path is already mounted for ceph_fuse
+  while ((mnt = getmntent(mtab)) != NULL) {
+    if ((strcmp(mnt->mnt_fsname, _CEPH_FUSE) == 0)
+	&& (strcmp(mnt->mnt_dir, path) == 0)
```

Comment:
Possible you should comare the `mnt_opts` too ? If the `mnt_opts` are different you should remount it by updating the options ?

Reply:
I don't know FUSE that well, but I think it may not allow for "-o remount". From the kernel driver portion:

```
        if (fsc->purpose == FS_CONTEXT_FOR_RECONFIGURE) {
                /*
                 * Ignore options coming from mount(MS_REMOUNT) for backward
                 * compatibility.
                 */
                if (fsc->oldapi)
                        return 0;

                return invalfc(fsc, "No changes allowed in reconfigure");
        }
```

Trying to apply new options to an existing mount sounds difficult to get right. I probably wouldn't do that here.